diff --git a/lib/Makefile b/lib/Makefile
index c8ac4d8..dc1cff9 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -65,6 +65,11 @@ ifeq ($(TARGET_OS), Darwin)
 	SHARED_EXT_MAJOR = $(LIBVER_MAJOR).$(SHARED_EXT)
 	SHARED_EXT_VER = $(LIBVER).$(SHARED_EXT)
 	SONAME_FLAGS = -install_name $(libdir)/liblz4.$(SHARED_EXT_MAJOR) -compatibility_version $(LIBVER_MAJOR) -current_version $(LIBVER) -dynamiclib
+else ifeq ($(TARGET_OS), OS/390)
+	SHARED_EXT = so
+	SONAME_FLAGS = -oliblz4.$(SHARED_EXT).$(LIBVER_MAJOR)
+	SHARED_EXT_MAJOR = $(SHARED_EXT).$(LIBVER_MAJOR)
+	SHARED_EXT_VER = $(SHARED_EXT).$(LIBVER)
 else
 	ifeq ($(WINBASED),yes)
 		SHARED_EXT = dll
@@ -131,7 +136,7 @@ else   # not windows
 $(LIBLZ4): $(SRCFILES)
   ifeq ($(BUILD_SHARED),yes)
 	@echo compiling dynamic library $(LIBVER)
-	$(CC) $(ALLFLAGS) -shared $^ -fPIC -fvisibility=hidden $(SONAME_FLAGS) -o $@
+	$(CC) $(ALLFLAGS) -shared $^ -fPIC -fvisibility=hidden $(SONAME_FLAGS) ${LIBS} -o $@
 	@echo creating versioned links
 	$(LN_SF) $@ liblz4.$(SHARED_EXT_MAJOR)
 	$(LN_SF) $@ liblz4.$(SHARED_EXT)
