diff --git a/lib/Makefile b/lib/Makefile
index c12949b..efaddc9 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -64,6 +64,11 @@ ifeq ($(TARGET_OS), Darwin)
 	SHARED_EXT_MAJOR = $(LIBVER_MAJOR).$(SHARED_EXT)
 	SHARED_EXT_VER = $(LIBVER).$(SHARED_EXT)
 	SONAME_FLAGS = -install_name $(libdir)/liblz4.$(SHARED_EXT_MAJOR) -compatibility_version $(LIBVER_MAJOR) -current_version $(LIBVER)
+else ifeq ($(TARGET_OS), OS/390)
+	SHARED_EXT = so
+	SONAME_FLAGS = -oliblz4.$(SHARED_EXT).$(LIBVER_MAJOR)
+	SHARED_EXT_MAJOR = $(SHARED_EXT).$(LIBVER_MAJOR)
+	SHARED_EXT_VER = $(SHARED_EXT).$(LIBVER)
 else
 	SONAME_FLAGS = -Wl,-soname=liblz4.$(SHARED_EXT).$(LIBVER_MAJOR)
 	SHARED_EXT = so
