diff --git a/tests/Makefile b/tests/Makefile
index 47e2774..e4a1a2c 100644
--- a/tests/Makefile
+++ b/tests/Makefile
@@ -36,9 +36,6 @@ PYTHON  ?= python3
 DEBUGLEVEL?= 1
 DEBUGFLAGS = -g -DLZ4_DEBUG=$(DEBUGLEVEL)
 USERCFLAGS:= -O3 $(CFLAGS) # appended for higher priority
-WFLAGS    = -Wall -Wextra -Wundef -Wcast-qual -Wcast-align -Wshadow \
-            -Wswitch-enum -Wdeclaration-after-statement -Wstrict-prototypes \
-            -Wpointer-arith -Wstrict-aliasing=1
 CFLAGS    = $(WFLAGS) $(DEBUGFLAGS) $(USERCFLAGS)
 CPPFLAGS += -I$(LIBDIR) -I$(PRGDIR) -DXXH_NAMESPACE=LZ4_
 ALLFLAGS  = $(CFLAGS) $(CPPFLAGS) $(LDFLAGS)
